### ISSUE TITLE
[BUG] ORC predicate pushdown should work with case-insensitive analysis

### ISSUE THREAD

**Issue by razajafri** [NVIDIA/spark-rapids#961]
commit 6c88d7c1259ea9fe89f5c8190c683bba506d528e
Author: Liang-Chi Hsieh <viirya@gmail.com>
Date:   Tue Aug 25 04:42:39 2020 +0000

    [SPARK-32646][SQL][3.0][TEST-HADOOP2.7][TEST-HIVE1.2] ORC predicate pushdown should work with case-insensitive analysis

**Comment by kuhushukla** [comment]
Can we name this issue with some detail on what it is about

**Comment by jlowe** [comment]
Agree with @kuhushukla the headline is not helpful, even after it was updated.  Please make the headline something much more descriptive about what this is tracking.  In general issue headlines with "this" and "we" in the title aren't very helpful because readers don't have the context to map the pronouns.

**Comment by kuhushukla** [comment]
@razajafri Could you elaborate on why and how we think this needs to be covered and adjust the priority if needed?

**Comment by razajafri** [comment]
Predicate push down for Orc seems to be broken regardless of the flag `spark.sql.caseSensitive`. I have tried the following query with Spark 3.0.0 and Spark 3.0.1 with 0.2.0 and 0.3.0 version of the plugin.

```
      val table = """CREATE TABLE `test_orc_data` (
      `_col1` INT,
      `_col2` STRING,
      `_col3` INT)
      USING orc"""
    
    spark.sql(table).collect
    
    spark.sql("insert into test_orc_data values(13, '155', 2020)").collect
    
    val df = """select _col2 from test_orc_data where _COL2 = 155"""
    spark.sql(df).collect
```

It works fine when `using parquet`. 

Looks like this hasn't worked since 0.2.0 at least.

**Comment by jlowe** [comment]
I was able to reproduce the issue, but it disappears when the fix from #1240 is applied.  I suspect this is related solely to the problem of ORC files without column names (or names that appear to be the default names, i.e.: using the `_col` prefix).
